applications/vanilla/library/Jobs/ExecuteBatchDeleteDiscussion.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * @copyright 2009-2020 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+use Garden\Schema\Schema;
+use Vanilla\Scheduler\Job\JobExecutionStatus;
+use Vanilla\Scheduler\Job\JobPriority;
+
+/**
+ * Execute the discussion delete queue.
+ */
+class ExecuteBatchDeleteDiscussion implements Vanilla\Scheduler\Job\LocalJobInterface {
+
+    /** @var DiscussionModel */
+    private $discussionModel;
+
+    /** @var array */
+    private $message;
+
+    /**
+     * Initial job setup.
+     *
+     * @param DiscussionModel $discussionModel
+     */
+    public function __construct(DiscussionModel $discussionModel) {
+        $this->discussionModel = $discussionModel;
+    }
+
+    /**
+     * Validate the message against the schema.
+     *
+     * @return Schema
+     */
+    private function messageSchema(): Schema {
+        $schema = Schema::parse([
+            "discussionID" => [
+                [
+                    "type" => "integer"
+                ]
+            ]
+        ]);
+        return $schema;
+    }
+
+    /**
+     * Execute all queued up items in the ActivityModel queue.
+     */
+    public function run(): JobExecutionStatus {
+        if (!is_array($this->message)) {
+            return JobExecutionStatus::abandoned();
+        }
+        $this->discussionModel->deleteID($this->message);
```

Comment:
This method is documented as only taking an integer for its first parameter (before you changed the doc in this PR). Here, it looks like we're passing it an associative array (e.g. ["discussionID" => [1, 2, 3]]). Is this something we're doing intentionally and it's working as it's supposed to? Or is this happening to work by chance, even though it's unintentional? I wouldn't want this to break in the future because the method is changed, not accounting for unintended usage. Are there many other calls to `deleteID` in the codebase that pass in an array, instead of an integer?

Reply:
There isn't, but the idea was to pass an array instead of looping over an id. What do you suggest I do here?
It worked because sql where accepts an array.